src/Auth0Client.ts

```diff
@@ -610,6 +611,17 @@ export default class Auth0Client {
       scope: getUniqueScopes(this.defaultScope, this.scope, options.scope)
     };
 
+    return singlePromise(
+      () =>
+        this._getTokenSilently({
+          ignoreCache,
+          getTokenOptions
+        }),
+      `${this.options.client_id}::${getTokenOptions.audience}::${getTokenOptions.scope}`
+    );
+  }
+
+  private async _getTokenSilently({ ignoreCache, getTokenOptions }) {
```

Comment:
Moved the original code to get a token to its own private method in order to be able to easily wrap it.